packages/packages/core/editor-components/src/utils/get-component-ids.ts

```diff
@@ -24,13 +27,15 @@ export const getComponentIds = async ( elements: V1ElementData[] ) => {
 		}
 
 		if ( !! childElements?.length ) {
-			ids.push( ...( await getComponentIds( childElements ) ) );
+			const newIds = await getComponentIds( childElements );
+
+			ids.push( ...( newIds.filter( Boolean ) as number[] ) );
 		}
 
 		return ids;
 	} );
 
 	const result = ( await Promise.all( components ) ).flat();
 
-	return Array.from( new Set( result ) );
+	return Array.from( new Set( result ) ).filter( Boolean ) as number[];
```

Comment:
i know id will probably never be 0 but still i think it's safer (and also no need to handle the type issue)
```suggestion
	return Array.from( new Set( result ) ).filter( ( value ) => value !== undefined );
```